Yosemite/Yosemite/Actions/CardPresentPaymentAction.swift

```diff
@@ -4,6 +4,31 @@
 import Combine
 
 public enum CardPresentPaymentAction: Action {
+    /// Switches the store to use WCPay as the backend. This is also the default.
+    ///
+    case useWCPay
```

Comment:
I don't mind declaring these two cases. It makes the code extremely declarative. 

One potential way to make this a bit more abstract could be to declare an enumeration (i.e. PaymentGateway) with two cases, and rename these two actions into a single action: `useGateway(gateway: PaymentGateway)

It could kind of future proof the API for a potential third gateway. On the other hand, a third gateway does not sound like something we would add in the nearly future, so 🤷🏼

Reply:
Right. There is one more argument supporting doing this soon, without waiting for a third plugin: serializing an enumeration to user defaults is simple. Have in mind we are going to have to save the plugin selected by the user to user defaults, and materialize it again from user defaults before performing any other operations. We could just save and read a boolean flag, but abstracting that behind an enumeration is very low effort for a significant benefit, imo.

So, sure, let's defer it, but not for too long.